app/assets/javascripts/jquery_overrides.js

```diff
@@ -37,10 +37,20 @@ $.ajaxSetup({
     }),
     "text script": logError(function (text) {
       if (text.match(/^{/)) {
+        console.debug('json payload');
         return jQuery.jsonPayload(text, function (text) {
           return text;
         });
+
+      } else if (text.match(/<!DOCTYPE html>/)) { // plain HTML payload fallback
+        console.debug('html payload');
+        document.open();
+        document.write(text);
```

Comment:
Hm.. should we be replacing the whole page?

I was thinking something like `$('#magic_div').html(text)` .. or is that the next step?